packages/shr-fhir-export/lib/extensions.js

```diff
@@ -538,6 +552,54 @@ class ExtensionExporter {
     }
   }
 
+  inlineSubextension(subExt, extension, field, baseId, baseExt) {
+    for (const ext of subExt.snapshot.element) {
+      const snapExt = common.cloneJSON(ext);
+      // Slice off the extraneous "Extension" that will start the id
+      const splitPoint = snapExt.id.indexOf('.');
+      const endId = snapExt.id.slice(splitPoint + 1);
+      // Construct a newId and path for the inlined version of the extension
+      const newId = `${baseId}.extension:${common.shortID(field.effectiveIdentifier)}.${endId}`;
+      const pathEnd = endId.slice(endId.lastIndexOf('.') + 1);
+      const existingElement = extension.snapshot.element.find(el => el.id === newId);
+      // Applying constraints can add elements that we don't actually want, just keep values
```

Comment:
Can you provide an example of an element that's added that we don't want?  You don't have to add it in the code -- I just want an example for my review.  I'm having trouble picturing it.

Reply:
So for exapmle on the `ReferenceRange` extension, there is a `Type` subextension. That `Type` subextension is bound to a valueSet on `ReferenceRange`. But that binding is only on `Type` in the context of `ReferenceRange`, not on the `Type` extension in general. The addition of this binding is done in the `applyConstraints` function which is called before `inlineSubextension`. To add this binding, we do a pseudo-inlining of the element onto the snapshot element. You can see how this used to look before inlining below.
![Capture](https://user-images.githubusercontent.com/52747882/66847644-bdef4c80-ef41-11e9-9f3e-b5517e1c3cf4.PNG)
Those `id` and `url` elements on `Type` in the picture above are extraneous, and we want to overwrite them. These are the elements that I am removing with this code. Values are not removed, because we do want to keep the bindings or fixed values that are created in `applyConstraints`.